egs2/dral/s2st1/local/prepare_datasets.py

```diff
@@ -0,0 +1,391 @@
+        elif archive_path.name.endswith(".tar.gz"):
+            subfolder = remove_suffix(archive_path.name, ".tar.gz")
+        else:
+            subfolder = archive_path.name
+        return destination_path / subfolder
+
+
+def download_thread_helper(url: str, dest_dir: Path) -> Tuple[str, Path]:
+    dataset_helper = DatasetHelper()
+    with tempfile.TemporaryDirectory() as tmp_dir_str:
+        tmp_dir: Path = Path(tmp_dir_str)
+        tar_gz_path: Path = dataset_helper.download(url, tmp_dir / Path(url).name)
+        return url, dataset_helper.uncompress(tar_gz_path, dest_dir)
+
+
+class DatasetConfig(BaseModel):
+    dataset_urls: List[str]
+
+
+def _write_kaldi_files(
+    raw_data_dir: Path,
+    data_file_stem: str,
+    df: pd.DataFrame,
+    src_lang: str,
+    tgt_lang: str,
+    output_dir: Path,
+):
+    src_lang = src_lang.lower()
+    tgt_lang = tgt_lang.lower()
+    LOGGER.info(f"writing kaldi-style files: {data_file_stem} -> {output_dir}")
+    assert data_file_stem in _AUDIO_FOLDERS, data_file_stem
+    output_dir.mkdir(exist_ok=True, parents=True)
+    # wav.scp
+    df["audio_path"] = df["id_recordings"].apply(
+        lambda x: str(raw_data_dir / data_file_stem / f"{x}.wav")
+    )
+    df[["id_recordings", "audio_path"]].to_csv(
+        output_dir / f"wav.scp.{src_lang}", sep=" ", header=False, index=False
+    )
+    df["trans_audio_path"] = df["trans_id"].apply(
+        lambda x: str(raw_data_dir / data_file_stem / f"{x}.wav")
+    )
+    df[["trans_id", "trans_audio_path"]].to_csv(
+        output_dir / f"wav.scp.{tgt_lang}", sep=" ", header=False, index=False
+    )
+    # utt2spk
+    if data_file_stem in {_RECORDINGS, _FRAGMENTS_LONG}:
+        df["spk"] = df.apply(
+            lambda x: "-".join(
+                sorted(
+                    [
+                        str(x["participant_id_left_unique"]),
+                        str(x["participant_id_right_unique"]),
+                    ]
+                )
+            ),
+            axis=1,
+        )
+    else:
+        df["spk"] = df["participant_id_unique"]
+    src_utt2spk = df[["id_recordings", "spk"]]
+    tgt_utt2spk = df[["trans_id", "spk"]]
+    tgt_utt2spk.columns = src_utt2spk.columns
+    utt2spk = pd.concat([src_utt2spk, tgt_utt2spk], axis=0, ignore_index=True)
+    utt2spk.to_csv(output_dir / "utt2spk", sep=" ", header=False, index=False)
+    # spk2utt
+    # Group by 'spk' and collect 'id' values into sets
+    grouped_data = utt2spk.groupby("spk")["id_recordings"].apply(set).reset_index()
+    with (output_dir / "spk2utt").open("w") as spk2utt_out:
+        for _, row in grouped_data.iterrows():
+            spk = row["spk"]
+            utt_str = " ".join(sorted(list(row["id_recordings"])))
+            spk2utt_out.write(f"{spk} {utt_str}\n")
+    # segments
+    if data_file_stem == _RECORDINGS:
+        df[["id_fragments", "id_recordings", "time_start", "time_end"]].to_csv(
```

Comment:
To double check, is segments working in the current implementation of s2st? I did not fully check this function at the time cause cvss is segmented.